app/components/school-vocabularies-list.js

```diff
@@ -1,101 +1,85 @@
-    toggleShowNewVocabularyForm() {
-      this.set('newVocabularyTitle', null);
-      this.set('showNewVocabularyForm', !this.showNewVocabularyForm);
-    },
+  @restartableTask
+  *load() {
+    this.vocabulariesRelationship = yield this.args.school.vocabularies;
+  }
 
-    confirmRemoval(vocabulary) {
-      this.set('showRemovalConfirmationFor', vocabulary);
-    },
-
-    cancelRemoval() {
-      this.set('showRemovalConfirmationFor', null);
-    }
-  },
+  @action
+  toggleShowNewVocabularyForm() {
+    this.newVocabularyTitle = null;
+    this.showNewVocabularyForm = !this.showNewVocabularyForm;
+  }
 
-  keyUp(event) {
-    const keyCode = event.keyCode;
-    const target = event.target;
+  @action
+  confirmRemoval(vocabulary) {
+    this.showRemovalConfirmationFor = vocabulary;
+  }
 
-    if ('text' !== target.type) {
-      return;
-    }
+  @action
+  cancelRemoval() {
+    this.showRemovalConfirmationFor = null;
+  }
 
+  @dropTask
+  *saveOrCancel(event) {
+    const keyCode = event.keyCode;
     if (13 === keyCode) {
-      this.saveNew.perform(this.newVocabularyTitle);
-      return;
-    }
-
-    if (27 === keyCode) {
-      this.send('toggleShowNewVocabularyForm');
+      yield this.saveNew.perform();
+    } else if (27 === keyCode) {
+      this.args.close();
     }
-  },
+  }
 
-  saveNew: task(function* (title) {
-    this.send('addErrorDisplayFor', 'newVocabularyTitle');
-    const { validations } = yield this.validate();
-    if (validations.get('isValid')) {
-      const school = this.school;
-      const vocabulary = this.store.createRecord('vocabulary', {title, school, active: true});
-      const savedVocabulary = yield vocabulary.save();
-      const vocabularies = yield school.get('vocabularies');
-      vocabularies.pushObject(savedVocabulary);
-      this.send('removeErrorDisplayFor', 'newVocabularyTitle');
-      this.set('showNewVocabularyForm', false);
-      this.set('newVocabularyTitle', null);
-      this.set('newVocabulary', savedVocabulary);
+  @dropTask
+  *saveNew() {
+    this.addErrorDisplaysFor(['newVocabularyTitle']);
+    const isValid = yield this.isValid();
+    if (! isValid) {
+      return false;
     }
-  }).drop(),
+    const vocabulary = this.store.createRecord('vocabulary', {
+      title: this.newVocabularyTitle,
+      school: this.args.school,
+      active: true
+    });
+    const savedVocabulary = yield vocabulary.save();
+    const vocabularies = yield this.args.school.get('vocabularies');
+    vocabularies.pushObject(savedVocabulary);
+    this.clearErrorDisplay();
+    this.showNewVocabularyForm =  false;
+    this.newVocabularyTitle =  null;
+    this.newVocabulary = savedVocabulary;
+  }
 
-  remove: task(function* (vocabulary) {
-    const school = this.school;
-    const vocabularies = yield school.get('vocabularies');
+  @dropTask
+  *remove(vocabulary) {
+    const vocabularies = yield this.args.school.get('vocabularies');
     vocabularies.removeObject(vocabulary);
```

Comment:
I believe this can be removed and that destroying the record will also remove this relationship and update the array in the store.

Reply:
confirmed, and removed.